sample/src/main/java/com/example/kyrie/KotlinDemoFragment.kt

```diff
@@ -0,0 +1,296 @@
+import android.os.Bundle
+import android.support.annotation.ColorInt
+import android.support.v4.app.Fragment
+import android.support.v4.content.ContextCompat
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.github.alexjlockwood.kyrie.Animation
+import com.github.alexjlockwood.kyrie.KyrieDrawable
+import com.github.alexjlockwood.kyrie.PathData
+import com.github.alexjlockwood.kyrie.StrokeLineCap
+import com.github.alexjlockwood.kyrie.asPath
+import com.github.alexjlockwood.kyrie.asPathData
+import com.github.alexjlockwood.kyrie.asPathInterpolator
+import com.github.alexjlockwood.kyrie.circle
+import com.github.alexjlockwood.kyrie.group
+import com.github.alexjlockwood.kyrie.kyrieDrawable
+import com.github.alexjlockwood.kyrie.path
+import com.github.alexjlockwood.kyrie.pathDataAnimation
+import com.github.alexjlockwood.kyrie.withListener
+import kotlinx.android.synthetic.main.fragment_choice.*
+import java.util.*
+
+class KotlinDemoFragment : Fragment() {
+
+    private val polygons = arrayOf(
+        Polygon(15, -0x17b39b, 362f, 2),
+        Polygon(14, -0x17b39b, 338f, 3),
+        Polygon(13, -0x2aab27, 314f, 4),
+        Polygon(12, -0x509112, 292f, 5),
+        Polygon(11, -0xb5b51a, 268f, 6),
+        Polygon(10, -0xbd6b19, 244f, 7),
+        Polygon(9, -0x941112, 220f, 8),
+        Polygon(8, -0xbd186c, 196f, 9),
+        Polygon(7, -0xa518a6, 172f, 10),
+        Polygon(6, -0x521895, 148f, 11),
+        Polygon(5, -0x101045, 128f, 12),
+        Polygon(4, -0x186bbe, 106f, 13),
+        Polygon(3, -0x17b39b, 90f, 14)
+    )
+
+    private lateinit var listener: SampleListenerAdapter
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.fragment_choice, container, false)
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        listener = SampleListenerAdapter(seekbar)
+        buttonHeart.setOnClickListener { startDrawable(createHeartDrawable()) }
+        buttonProgCirc.setOnClickListener { startDrawable(createCircularDrawable()) }
+        buttonProgLine.setOnClickListener { startDrawable(createHorizontalDrawable()) }
+        buttonPolygon.setOnClickListener { startDrawable(createPolygonDrawable()) }
+        buttonMorph.setOnClickListener { startDrawable(createMorphDrawable()) }
+    }
+
+    private fun startDrawable(drawable: KyrieDrawable) {
+        imageView.setImageDrawable(drawable withListener listener)
+        drawable.start()
+    }
+
+
+    private fun createHeartDrawable(): KyrieDrawable {
+        return context?.let {
+            KyrieDrawable.create(it, R.drawable.avd_heartbreak)
+        } ?: error("No context available")
+    }
+
+    private fun createHorizontalDrawable(): KyrieDrawable {
+        return kyrieDrawable {
+            viewport = 360f x 10f
```

Comment:
hm, it looks very nice but the fact that `x` is available on arbitrary `Float` and returns very specific type kinda worries me

Is this too bad?:

```kotlin
kyrieDrawable {
    viewPort = Size(360, 10)
}
```

or a lowercase function/type-name

```kotlin
kyrieDrawable {
    viewPort = size(360, 10)
}
```

Reply:
specifically because there are lots of other calls like `fillAlpha(0.3f)` and infix function looks kinda weird around them, but maybe it's just me :)

Reply:
Wasn't sure about the 'x' function. I know Romain Guy used that as an example of things they don't want in KTX. So I'll just replace it by a simple size() function.
Anyway, it wasn't available for any Float object: the Float extension was declared inside the KyrieDrawableDsl class and was only available within that scope.

Reply:
hm, I'm unaware of this scoping behavior, will need to double-check